I need feedback on what I've done for a Get Involved page.

jerrac opened the pull request:
Here is a very rough outline of a Get Involved page. I've stubbed in some content, and I've added some sections without any content.

I'm not sure if I'm using the `{{ _() }}` syntax correctly. Specifically where I have inline links. Can I put the links inside the jinja syntax?

Do we actually need every single section I've added?

Sancus commented:
I'll review this tomorrow.

ryanleesipes commented:
I'm going to pull this down and take a look today.

Sancus commented on website/get-involved/index.html:
```diff
@@ -0,0 +1,172 @@
+                </section>
+                <section>
+                    <h3>{{ _('Getting Started') }}</h3>
+                    <p>{{ _('Read the') }} <a href="https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Introduction">{{ _('Developer Guide') }}</a> {{ _('on MDN. That will give you a general overview of the ecosystem Thunderbird is part of.') }}</p>
```
In general, one sentence should be one gettext call, beccause localizers can't effectively translate a bunch of isolated sentence fragments. <a> tags can be included in the localized text, and should be so, localizers have some context for what the link is. See: https://github.com/thundernest/thunderbird-website/blob/master/website/features/index.html#L206

If the whole sentence is a link, like the brief phrases on the front page describing links underneath the download button, it's OK to exclude the <a> tag like: https://github.com/thundernest/thunderbird-website/blob/master/website/_includes/download-button.html#L57

jerrac replied:
Thanks, I was wondering about that. It did seem kinda silly to do it the way I did. :)

Sancus commented on website/get-involved/index.html:
```diff
@@ -0,0 +1,172 @@
+{% block page_title %}{{ _('Get Involved') }}{% endblock %}
+
+{% block page_desc %}
+{% if LANG.startswith('en-') %}
```
Not sure why this is here, the whole page will be en-US only.

jerrac replied:
It was left over from my paste of the features page. I likely left it in because I remembered your comment saying I should stick the gettext calls in, but forgot you had said to leave off that bit.

Sancus commented on .gitignore:
```diff
@@ -4,3 +4,4 @@ product-details-json
 thunderbird_notes
 thunderbird.net
 site
+.idea/*
```
I'm not sure what this is for but it doesn't seem like something that should be committed.

jerrac replied:
I use Jetbrains Phpstorm IDE. It uses .idea/* directories to store settings. So I added it to the .gitignore file. I figured it shouldn't hurt anything.

I used a separate commit so it could be dropped easily. So I'll make sure it isn't in the pull request when I update.

Sancus commented:
There aren't that many links yet but 8 sections are roughed out(Development, Documentation, Communication, Infrastructure, Design, Translation, Quality Assurance, Marketing) and I given that Development already has like 7 and two subsections, I think it's easy to see how this could eventually expand to 50+. That's why I was just cautioning to keep it as brief as possible.

ryanleesipes commented:
Reviewed, @jerrac this is a great start! Fantastic work!

@wsmwk here are a couple of screenshots. I'll point Ura Design at this and see if they have any feedback, they have some pretty awesome ideas for the site altogether, they will be sending a PR soon, so I'll make sure they are aware we are working on this.
![screenshot-get-involved](https://user-images.githubusercontent.com/2141621/37789909-1f2be562-2dd3-11e8-9dc1-db4b3d0d96ac.png)
![screenshot-get-involved-2](https://user-images.githubusercontent.com/2141621/37789910-1f4b6734-2dd3-11e8-883b-78f37c1d4e64.png)

ryanleesipes commented:
Attempt to solve #19 (linking for the sake of others who are trying to keep up).

ryanleesipes commented:
@jerrac can you flip the "Getting Started" and "Code" headings under development? I think having "Getting Started" first flows better. What do you think?

jerrac commented:
Ok, I've updated. And here's a screenshot of the whole page. 
![screenshot-localhost 8001-2018-03-24-12-41-46](https://user-images.githubusercontent.com/910764/37868278-feeee094-2f60-11e8-9f18-6554ef68f16e.png)

jerrac commented:
> @Sancus In general I feel like this looks like it might end up too long.

I pulled the sections from LibreOffice's version of the page. And added "Communication" myself since that's a section I'd really appreciate being filled out myself.

If there's a section we don't want, we can just drop it. For example, I'm not sure how Infrastructure or Marketing would apply to Thunderbird. 

> I'm moderately concerned that a big page with 50-100+ links will end up poorly maintained and put people off as well. I don't think it's necessarily bad to have each of these sections, but as you are filling them out, please keep in mind that you don't want to infodump too much on the first page people look at.

If we make sure the content is almost all "This is where you go to find out how to get involved with that." I think we should be ok. 

Anyway, for some more content, can you help answer these questions? (Some answers I can find myself, but I'm listing them here so I don't forget...)

* What other hg.mozilla.org repositories are associated with Thunderbird?
* Are there any other repositories on GitHub, GitLab, Bitbucket, etc?
* I should add information/link about extension development
* Is there a specific sub section of MDN that contains all the Thunderbird related docs?
* Where is information on how to contribute to MDN Thunderbird docs?
* Is there anything specific about contributing to the Thunderbird section of mozilla wiki that isn't covered by https://wiki.mozilla.org/MozillaWiki:Policies/Accounts ?
* I saw a list of IRC channels somewhere, but I need to find it again.
* Need to find the list of mailing lists...
* Are there any Google Groups?
* Is infrastructure handled entirely by Mozilla? Can we just drop that section? (I'm think of managing the thunderbird.net server, an build servers, etc...)
* I'm not sure I've seen anything about how to contribute design work, can you point me in the right direction?
* Is the current MDN link all we need for Translation?
* Maybe we should just rewrite the https://wiki.mozilla.org/Thunderbird:Testing page and replace it with the QA section? 
* I have no idea what to put in for Marketing. Should I just drop it?

ryanleesipes commented:
My answers to your questions:

**What other hg.mozilla.org repositories are associated with Thunderbird?**
comm-central is the most important here, we'll stick to that for now.

**Are there any other repositories on GitHub, GitLab, Bitbucket, etc?**
There are some other repos here: https://github.com/mozilla-comm (eventually I hope to marry Thundernest and mozilla-comm).

**I should add information/link about extension development**
https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Thunderbird_extensions <- point them here to get started.

**Is there a specific sub section of MDN that contains all the Thunderbird related docs?**
There is! It is here: https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird

**Where is information on how to contribute to MDN Thunderbird docs?**
Log in via GitHub on the MDN site and hit the edit button. This is a "Getting Started" Guide for MDN: https://developer.mozilla.org/en-US/docs/MDN/Getting_started

**Is there anything specific about contributing to the Thunderbird section of mozilla wiki that isn't covered by https://wiki.mozilla.org/MozillaWiki:Policies/Accounts ?**
No, that is fine.

**I saw a list of IRC channels somewhere, but I need to find it again.**
https://wiki.mozilla.org/Thunderbird/CommunicationChannels - the maildev channel is listed here, there is also a channel called #thunderbird on the same network. #maildev is for hacking on Thunderbird, #thunderbird is for general questions and support. There is #instantbird for chat.

**Need to find the list of mailing lists...**
Multiple mailing lists listed on https://wiki.mozilla.org/Thunderbird/CommunicationChannels

**Are there any Google Groups?**
Not to my knowledge.

**Is infrastructure handled entirely by Mozilla? Can we just drop that section? (I'm think of managing the thunderbird.net server, an build servers, etc...)**
We handle it, Mozilla still hosts some stuff too. We can leave this off.

**I'm not sure I've seen anything about how to contribute design work, can you point me in the right direction?**
For now we'll ask them to file design ideas and concepts as enhancements: https://bugzilla.mozilla.org/enter_bug.cgi?product=Thunderbird&bug_severity=enhancement - they can also join #maildev or post concepts on the forum: https://discourse.mozilla.org/c/thunderbird

**Is the current MDN link all we need for Translation?**
Maybe @wsmwk or @kewisch will have knowledge of any other translation resources we should include. As you pointed out, our localization on the Wiki points at the MDN page: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Thunderbird/Thunderbird_Localization

**Maybe we should just rewrite the https://wiki.mozilla.org/Thunderbird:Testing page and replace it with the QA section?**
Not opposed to this, make an up-to-date version.

**I have no idea what to put in for Marketing. Should I just drop it?**
For now, just drop it. We might bring it back, but I'm currently talking to folks in the community about starting a marketing team, and we've got a ways to go before we start linking to resources.

Sancus commented:
There is a support group that sees some traffic: https://groups.google.com/forum/#!forum/mozilla.support.thunderbird

tb-planning(which is listed on the communications wiki above) is also mirrored to a group: https://groups.google.com/forum/#!forum/tb-planning

jerrac commented:
Thanks @ryanleesipes and @Sancus for the content!

I'd say it's looking pretty complete now. What am I missing?

Here's an updated screenshot:

![screenshot-thunderbird net-get-involved-page-2018-04-08-13-38-39](https://user-images.githubusercontent.com/910764/38472277-6e1af002-3b32-11e8-9f13-6ef358b08c2f.png)

wsmwk commented:
I'm coming to like to monolitihic page, and it's looking good.  The theme looks similar to http://thunderbird.net/ and although it might benifit from some graphics, I wouldn't hold up shipping it just for graphics. 

Can we shorten the tag line?  Like "Everything you need to get involved with Thunderbird"

Development > Code second sentence is awkward and can be combined with the first into "Read ... for a general overview of the ecosystem surrounding Thunderbird."  or "the Mozilla ecosystem of which Thunderbird is a part(icpant?)"

Simple TB build ... c/is a good place to find how to/describes how to/

For search purposes we need to word "add-on" somewhere in the document.

Documentation is missing line items for SUMO KB.

Communication > Forums can we add 
* discourse link for Thunderbird beta
* mozilla.dev.apps.thunderbird newsgroup (the mirror to dev-apps-thunderbird)

Add to "Thunderbird builds on Mozillazine" ... for help on building Thunderbird and using daily buids.

dev-apps-thunderbird suggest something along the lines "For technical and code discussions, such as add-ons.

ryanleesipes commented:
Firstly, thanks @jerrac - this has really taken shape. I think we should probably merge this sooner rather than later, and then make additions and fixes if nothing is really wrong here.

We should probably add "Get Involved" to the top menu. Also, if it is easy enough, we could implement anchors to each part of this page for quick linking to the different sections (but this is more of a wish than a must have).

In response to @wsmwk's feedback

I agree on shortening the tagline.

"Read for a general overview of the Mozilla ecosystem of which Thunderbird is based on." is another possibility for making that sentence read better. But "of which Thunderbird is a part" or "participant" also sounds great.

@wsmwk what do you think the SUMO KB link should say? "Get support for Thunderbird"? "Help support Thunderbird users"?

What's the link to mozilla.dev.apps.thunderbird?

mkmelin commented:
Some comments/suggestions:

- Please  make the irc and newgroups real links (irc:// and nntp://)
- The code section should explan how to get the code, since that's not straight forward - see https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Build_Instructions/Simple_Thunderbird_build
- Good first bugs: easy easy -> good bugs to try to fix when you're starting out
- Under the Bugs section, make it clear handling incoming bugs is an excellent way to help out. Do link to "todays/recent bugs". A query like https://bugzilla.mozilla.org/buglist.cgi?list_id=14100150&chfield=%5BBug%20creation%5D&chfieldfrom=-3d&chfieldto=Now&query_format=advanced&product=MailNews%20Core&product=Thunderbird
- Under LIve Chat: #instandbird is about the... well I'm not really sure now. Instantbird the product is dead, but it's confusing for someone new to have a description "for chatting". About what?
- tb-planning: don't put the google mirror as a separate item, put the google mirror info in the same item, in parenthesis?
- dev-apps-thunderbird, that's just the mailing-list mirror for the mozilla.dev.apps.thunderbird newsgroup

wsmwk commented:
> @wsmwk what do you think the SUMO KB link should say? "Get support for Thunderbird"? "Help support Thunderbird users"?

For SUMO questions, "Support users asking for help in Thunderbird Forum"? https://support.mozilla.org/en-US/questions/thunderbird 

For SUMO KB "Help improve documentation"?  I don't have a link yet

> What's the link to mozilla.dev.apps.thunderbird?
nttp://mozilla.dev.apps.thunderbird   but my system apparently have Thunderbird registered as the application for nntp.  I wonder if that is true for other users?

jerrac commented:
@mkmelin

> Please make the irc and newgroups real links (irc:// and nntp://)
Do I need a specific port in the irc links? irc://irc.mozilla.org:<port>/thunderbird

I've been linking to the mail.mozilla.org/listinfo pages for the newsgroups. Should I replace those links with something like nntp://tb-planning ? What if the person isn't familiar with how newsgroups work, and is used to just subscribing to everything via a web page? (Like me.)

Would the nntp link actually be mail.mozilla.org/tb-planning?

> The code section should explan how to get the code, since that's not straight forward - see https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Build_Instructions/Simple_Thunderbird_build

I've linked to that help doc in the Development section. Actually decribing how to get the code seems a bit beyond the scope of this page. If it isn't, please tell me what I should put in.

> Under the Bugs section, make it clear handling incoming bugs is an excellent way to help out. 

Any suggestions on how to word that, and where to put it? (In the Bugs section of course, but where in the section? A subsection? Just above the current list?)

> Under Live Chat: #instandbird is about the... well I'm not really sure now. Instantbird the product is dead, but it's confusing for someone new to have a description "for chatting". About what?

That was a product? I had no idea. I thought it was just a general chat about whatever channel for people involved with Thunderbird to have fun in.

Should we just drop it?

> dev-apps-thunderbird, that's just the mailing-list mirror for the mozilla.dev.apps.thunderbird newsgroup

Is the Google Group the mirror, or is it the other way around? I found: https://lists.mozilla.org/listinfo/dev-apps-thunderbird and https://groups.google.com/forum/#!forum/mozilla.dev.apps.thunderbird Which should be the main link?

jerrac commented:
Here's a screenshot with the changes I've made so far in response to @wsmwk and @mkmelin .
![screenshot-thunderbird net-get-involved-page-2018-04-15-13-05-02](https://user-images.githubusercontent.com/910764/38782817-dc4154ce-40ad-11e8-977b-27e72a6080af.png)

jerrac commented:
Here's what the nav looks like with a get involved link.
![screenshot- thunderbird-nav-2018-04-15-13-16-16](https://user-images.githubusercontent.com/910764/38782888-4c463dd8-40af-11e8-9bc1-68d8e005e8a9.png)

mkmelin commented:
>     Please make the irc and newgroups real links (irc:// and nntp://)
>     Do I need a specific port in the irc links? irc://irc.mozilla.org:/thunderbird

No, irc://irc.mozilla.org/thunderbird should be fine

> I've been linking to the mail.mozilla.org/listinfo pages for the newsgroups. Should I replace those links with something like nntp://tb-planning ? What if the person isn't familiar with how newsgroups work, and is used to just subscribing to everything via a web page? (Like me.)

I think it's fine to link to the listinfo pages for the mailing lists, but when you mention the newsgroup, do link to it there.  There's no newsgroup for tb-planning.

>     The code section should explan how to get the code, since that's not straight forward - see https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Build_Instructions/Simple_Thunderbird_build
> 
> I've linked to that help doc in the Development section. Actually decribing how to get the code seems a bit beyond the scope of this page. If it isn't, please tell me what I should put in.

You could write that "Thunderbird also relies on the code from mozilla-central. See the build instructions for how to obtain all the code needed to build."

Came to mind, you might want to link to the searchable version of the code, since that's invaluable for someone looking to fix a bug. It's here: https://dxr.mozilla.org/comm-central/source/

>     Under the Bugs section, make it clear handling incoming bugs is an excellent way to help out.
> 
> Any suggestions on how to word that, and where to put it? (In the Bugs section of course, but where in the section? A subsection? Just above the current list?)

Don't think it needs a subsection. Just something like. 
"You can help out by triaging incoming bugs (link). Known bugs should be marked as duplicates, unconfirmed bugs often needs someone to see if they too can reproduce."

>     Under Live Chat: #instandbird is about the... well I'm not really sure now. Instantbird the product is dead, but it's confusing for someone new to have a description "for chatting". About what?
> 
> That was a product? I had no idea. I thought it was just a general chat about whatever channel for people involved with Thunderbird to have fun in.
> 
> Should we just drop it?

Probably just drop it yes. Much of the code is now in the instant messaging component of Thunderbird, I think devs do hang out in #maildev anyway.

>     dev-apps-thunderbird, that's just the mailing-list mirror for the mozilla.dev.apps.thunderbird newsgroup
>
> Is the Google Group the mirror, or is it the other way around? I found: https://lists.mozilla.org/listinfo/dev-apps-thunderbird and https://groups.google.com/forum/#!forum/mozilla.dev.apps.thunderbird Which should be the main link?

Google Groups is just a mirror. The list and newsgroup are, I think, both considered equal. Given that this page is targeting people with Thunderbird installed, it might be best to link to the newsgroup in this case.

wsmwk commented:
@Sancus If we go live now, with the expecation that we will be iterating over the next few weeks, do we inform l10n to hold off on their translation?

FWIW, https://www.thunderbird.net/en-US/calendar/ nicely illustrates what can be done in a compact format

It also lists bugsahoy which should be listed as https://www.joshmatthews.net/bugsahoy/?thunderbird=1

jerrac commented:
In the Development section I added lines about relying on mozilla-central. 

The build instructions link in that line goes to the main Build Instructions article on MDN. Would it be better for that link to hit the simple Thunderbird build article as well?

"There is a searchable DXR instance that you can use to search through the code." is what I said when linking to DXR. Is it ok as is, or should it be worded differently?

I added @mkmelin 's line about new bugs to the bugs section. I used the "Untreaged" link from https://bugzilla.mozilla.org/describecomponents.cgi?product=Thunderbird as the target. That should work, correct?

> ...but when you mention the newsgroup, do link to it there.

I think the only mentions of newsgroups are where the mailing list links are.

> Google Groups is just a mirror. The list and newsgroup are, I think, both considered equal. Given that this page is targeting people with Thunderbird installed, it might be best to link to the newsgroup in this case.

I still am not sure what the full URI of the newsgroup link should be. For now I'm just leaving as is. Maybe separate newsgroup links should be added for those lists that have them? Something like:

* listname (Newsgroup) (Google Group Mirror)

@wsmwk FYI, bugsahoy is linked to via the "Mentored Bugs" link. Does that meet what you were after?

New screenshot: 
![screenshot-thunderbird net-get-involved-page-2018-04-19-10-03-13](https://user-images.githubusercontent.com/910764/39006918-61ff9f9a-43b9-11e8-8f68-510f666a55f0.png)

mkmelin commented:
> The build instructions link in that line goes to the main Build Instructions article on MDN. Would it be better for that link to hit the simple Thunderbird build article as well?
> 
> "There is a searchable DXR instance that you can use to search through the code." is what I said when linking to DXR. Is it ok as is, or should it be worded differently?

Perhaps  "You can search the code on DXR."

> I added @mkmelin 's line about new bugs to the bugs section. I used the "Untreaged" link from https://bugzilla.mozilla.org/describecomponents.cgi?product=Thunderbird as the target. That should work, correct?

I thought I added a suitable link, but apparently not. I'd link to the incoming bugs for the 3 last days: https://bugzilla.mozilla.org/buglist.cgi?list_id=14114741&chfieldto=Now&query_format=advanced&chfield=%5BBug%20creation%5D&chfieldfrom=-3d&product=MailNews%20Core&product=Thunderbird

> I still am not sure what the full URI of the newsgroup link should be. For now I'm just leaving as is. Maybe separate newsgroup links should be added for those lists that have them? Something like:
> 
>     listname (Newsgroup) (Google Group Mirror)

WFM. Use news://news.mozilla.org/mozilla.dev.apps.thunderbird

I notice we're missing the maildev mailing list. 
http://lists.thunderbird.net/mailman/listinfo/maildev_lists.thunderbird.net
"Maildev is a moderated mailing list for Thunderbird Engineering plans" (no mirrors or newsgroup AFAIK)

Another thing that's missing is that there is no mention of continous integration. And no mention of automated tests (and how to write them). Not sure where that should be, but I think we want to include some information about it. https://treeherder.mozilla.org/#/jobs?repo=comm-central is where you can see it in action. https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Thunderbird_Automated_Testing is about the tests

wsmwk commented:
On 4/19/2018 3:34 PM, Magnus Melin wrote:
> Another thing that's missing is that there is no mention of continous 
> integration. And no mention of automated tests (and how to write 
> them). Not sure where that should be, but I think we want to include 
> some information about it. 
> https://treeherder.mozilla.org/#/jobs?repo=comm-central is where you 
> can see it in action. 
> https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Thunderbird_Automated_Testing 
> is about the tests

Given that (I assume) we don't have any doc about continuous integration 
I don't think there's much point to listing it.   If it isn't already 
listed on MDN it should be.

We should be wary of this page becoming a place to list everything 
Thunderbird.  (It seems headed that way)

mkmelin commented:
Well, it's extremely important for someone actually working on code (on trunk) to know where to see if it is going to build or not, and see recent checkins in a web interface. That's really major drivers to see activity and get sucked into the project.

mkmelin commented:
Somewhat OT, the "todays bugs" link MozillaZine used to(?) have was apparently an incredibly efficient on-boarding back in the day. I recall someone speaking about it at one of the summits, and I could *so* relate to that.

jerrac commented:
> > On 4/19/2018 3:34 PM, Magnus Melin wrote:
> >  Another thing that's missing is that there is no mention of continous
> >  integration. And no mention of automated tests (and how to write
> >  them). Not sure where that should be, but I think we want to include
> >  some information about it.
> >  https://treeherder.mozilla.org/#/jobs?repo=comm-central is where you
> >  can see it in action.
> >  https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Thunderbird_Automated_Testing
> >  is about the tests
> Given that (I assume) we don't have any doc about continuous integration
> I don't think there's much point to listing it.   If it isn't already
> listed on MDN it should be.

I'd think CI and Automated testing would be documented a step up from the Get Involved page. That's the kind of thing you'd look for once you want to push changes. IMO, it shouldn't be on the Get Involved page.


> We should be wary of this page becoming a place to list everything
> Thunderbird.  (It seems headed that way)

I agree. We should make sure everything on it has something to do with "getting involved." I'd almost want the communications section to have it's own page, but I know that when I was looking to get involved, that section would have really really helped me as it is.


Side note: The length of the page makes me wonder if getting involved with Thunderbird is just more complex that it needs to be. That's why I posted https://discourse.mozilla.org/t/consolidating-documentation-and-communication/27849

ryanleesipes commented:
@jerrac you raise good points on your thread on Discourse. I say we cut this off here and consider if we have much more that we want to document - that we look into improving documentation (i.e. the Wiki, a Comm-central README, etc). Which, I'd love to discuss anyway!

For now, I say we roll this in and then start a having a conversation about the rest of the documentation in #35.

jerrac commented:
@mkmelin http://lists.thunderbird.net/mailman/listinfo/maildev_lists.thunderbird.net sends me to a "Insecure site" warning in FF. Is that temporary?

Sancus commented:
> @Sancus If we go live now, with the expecation that we will be iterating over the next few weeks, do we inform l10n to hold off on their translation?

@wsmwk I can just make this page en-US only and keep it out of string extraction for now if you want. We don't have to localize new pages immediately, localization concerns are primarily for pages that are already localized(because of breaking translated strings that already exist).

Sancus commented:
There is supposed to be https but it looks like the cert has expired. Opened a ticket with the provider. We don't host the mailing list.

ryanleesipes commented:
LGTM.

jerrac commented:
> @Sancus but please remove the .idea bit from .gitignore.

Done.
